src/service/document-info-impl.js

```diff
@@ -92,6 +93,9 @@ export class DocInfo {
  * @param {!Window} win
  * @return {string}
  */
-function getPageViewId(win) {
-  return String(Math.floor(win.Math.random() * 10000));
+export function getPageViewId(win) {
+  if (!pageViewIdByWindow[win]) {
+    pageViewIdByWindow[win] = String(Math.floor(win.Math.random() * 10000));
```

Comment:
`win` is an object, so this evaluates to `pageViewIdByWindow["[object Window]"]` every time.